app/models/submission.rb

```diff
@@ -0,0 +1,21 @@
+class Submission < ApplicationRecord
+  self.table_name = "application"
+  has_many :submission_versions, dependent: :destroy, foreign_key: "application_id"
+
+  validates :application_state, presence: true
+  validates :application_type, presence: true
+  validates :application_risk, presence: true
+
+  def latest_version
+    # Optimisation: When pulling lists of submissions, this allows a single DB lookup using #includes,
+    # instead of going back to the database for every submission
+    submission_versions.max_by(&:version)
```

Comment:
Think can cheat slightly on this one.....

if you order the versions by `version: desc` in the include then you can just call first here and avoid running the ruby `max_by`

so... one option is a default scope on the versions table, but that sucks..

instead:
```
has_many :ordered_submission_versions, -> { order(version: :desc) }, dependent: :destroy, foreign_key: "application_id", table_name: 'SubmissionVersions'
```

and **I think** that will work (untested)

Reply:
Nice! I doubted it would work as `includes` is a big magical, but it does. Thanks!